graylog2-web-interface/src/pages/IndicesPage.jsx

```diff
@@ -48,4 +49,4 @@ class IndicesPage extends React.Component {
   }
 }
 
-export default IndicesPage;
+export default withHideOnCloud(IndicesPage);
```

Comment:
I suggest moving these checks into `AppRouter`, so that `react-router` will not have the route configuration for routes that should be hidden. This helps us by not having to modify each page that should be hidden, and the behaviour is more consistent with other routes that do not exist.

Something like this should do the trick:
```diff
--- a/graylog2-web-interface/src/routing/AppRouter.jsx
+++ b/graylog2-web-interface/src/routing/AppRouter.jsx
@@ -87,6 +87,7 @@ import {
 } from 'pages';
 
 import RouterErrorBoundary from 'components/errors/RouterErrorBoundary';
+import withHideOnCloud from '../util/conditional/withHideOnCloud';
 
 const AppRouter = () => {
   const pluginRoutes = PluginStore.exports('routes');
@@ -162,7 +163,7 @@ const AppRouter = () => {
                      component={EditContentPackPage} />
               <Route path={Routes.SYSTEM.CONTENTPACKS.show(':contentPackId')} component={ShowContentPackPage} />
               <Route path={Routes.SYSTEM.GROKPATTERNS} component={GrokPatternsPage} />
-              <Route path={Routes.SYSTEM.INDICES.LIST} component={IndicesPage} />
+              {withHideOnCloud(<Route path={Routes.SYSTEM.INDICES.LIST} component={IndicesPage} />)}
               <Route path={Routes.SYSTEM.INDEX_SETS.CREATE} component={IndexSetCreationPage} />
               <Route path={Routes.SYSTEM.INDEX_SETS.SHOW(':indexSetId')} component={IndexSetPage} />
               <Route path={Routes.SYSTEM.INDEX_SETS.CONFIGURATION(':indexSetId')}
```

Reply:
I was thinking the exact same thing earlier. Makes it a lot safer too. Will change the PRs to reflect this.